src/channels/message_sender.py

```
@@ -0,0 +1,303 @@
+            )
+            return {"success": success, "channel": "whatsapp"}
+        except Exception as e:
+            logger.error(f"WhatsApp send failed: {e}")
+            return {"success": False, "error": str(e), "channel": "whatsapp"}
+    
+    async def _send_discord_text(self, recipient: str, text: str, **kwargs) -> Dict[str, Any]:
+        """Send text message via Discord using Unix domain socket IPC."""
+        logger.info(f"_send_discord_text called: recipient={recipient}, text_length={len(text)}")
+        import aiohttp
+        import os
+        import re
```

Ruff (0.12.2)

`re` imported but unused

Remove unused import: `re`

(F401)